### ISSUE TITLE
\verbatim with "*/" within text block produces errors with clang assisted parsing

### ISSUE THREAD

**Issue by Shea690901** [doxygen/doxygen#9043]
**Describe the bug**
While
~~~
/**
@verbatim
some text followed by */
and some more text
@endverbatim
*/
~~~
or the equivalent with C++ line comments yields the expected html output it also gives clang errors (`CLANG_ASSISTED_PARSING = YES` in `Doxyfile`):
~~~
...
error: error.dox:4:5: error: C++ requires a type specifier for all declarations [clang]
error: error.dox:4:5: error: declaration of reference variable 'some' requires an initializer [clang]
error: error.dox:4:9: error: expected ';' after top level declarator [clang]
...
~~~

**Expected behavior**
No error messages by clang.
This is a verbatim text block and as such needs no parsing regardless of it's contents.
Btw.: the internal parser (`CLANG_ASSISTED_PARSING = NO`) doesn't give any errors...

**Screenshots**

**To Reproduce**
Upload doesn't work for me at the moment, but the above example source can be used to reproduce the error...
And as config a default config file with `CLANG_ASSISTED_PARSING = YES` may be used...

**Version**
1.9.1 (I currently can`t update to current release) on Linux:
~~~
LSB Version:    n/a
Distributor ID: Gentoo
Description:    Gentoo Base System release 2.8
Release:        2.8
Codename:       n/a
~~~

**Stack trace**

**Additional context**

PS.:
When using `@code{.unparsed}` instead of `@verbatim`the same error occures...

**Comment by albert-github** [comment]
The `.dox` file is mapped to be parsed as a `C / Cpp` file and thus in principle has to obey the `C / Cpp` rules.

I think that doxygen and Clang assisted parser are right. We have here an end of comment `*/` after the `\verbatim` and this terminates the comment and the rest will be seen as normal code by the CLang assisted parser and hence the warnings.

Why doesn't doxygen without Clang assisted parsing give a message? This is because doxygen has "some" knowledge about the meaning of the comment command and sees the `\verbatim` and decides "to skip to process " the rest of the file till it finds `\endverbatim` and thus ignores the `*/` inside the `\verbatim` part.

**Comment by Shea690901** [comment]
And you wrote yourself why doxygen with clang assisted parsing isn't right...

Doxygen itself knows about the meaning of @verbatim/@endverbatim and skips everything in between when parsing itself.
Just as expected by the meaning of "verbatim". And exactly the same expectations should be fulfilled with clang assisted parsing enabled...

Also if the '*/' within the verbatim block where to be seen as the end of the comment and not part of the text, than we had another error:
@verbatim without @endverbatim....
Because the is no '/**' after '*/' there would be no reason to see the existing @endverbatim as doxygen command....

**Comment by albert-github** [comment]
I think there is some misunderstanding here. 
The code like:
```
/**
  \verbatim
   */
  \endverbatim
*/
void fie();
```
is not correct Cpp / C code as the comment block ends at the first `*/` and the next statement (here: `\endverbatim`) is interpreted as a normal Cpp / C statement.

Why doesn't doxygen its own scanner flag this as an error?
This is due to historical reasons, in the past (before e.g. some markdown processing was possible) separate documentation had to be written in the form of Cpp / C comments as well.

Why does CLang interpreted flag this as an error?
The **entire** file that should be handled as a Cpp / C file is handed to CLang and as CLang has, understandably, only a limited knowledge about the doxygen commands is sees such a construct as incorrect.
With the proposed pull request #9073 (for issue #8784) some more control is possible.

I hope this explanation shines a bit of a light on the matter.

**Comment by Shea690901** [comment]
Then we don't need `@verbatim` or `@code{unparsed}`....
If soweit plain Dokumentation file, that is never(!) seen by a compiler, still has to be parsable by a C-compiler, despite telling doxygen _that doesn't make sense_,  then this is an unuasable feature and can be removed (what about `@code{.asm}`? Must this be parsable by a C-compiler too, because that doesn't make any sense either...)

**Comment by albert-github** [comment]
When you tell doxygen that a file has to be seen (e.g. by setting `EXTENSION_MAPPING = ddd=C` the file with extension `ddd` has to follow the rules for the specified language. When having `@code{.asm}` inside a "C" file then there cannot be be a `*/` in it either, for the already mentioned reason. Depending on the setting of the `EXTENSION_MAPPING` for the `asm` the code is shown as "plain" text or an attempt is made to do a code coloring according to the specified language.

**Comment by Shea690901** [comment]
Then why are plain documentation files treated as `C` by default and not as plain documentation (and using `EXTENSION_MAPPING = dox=md` doesn't work either, it's still parsed by `clang`)?
There is just no sense behind this..

**Comment by albert-github** [comment]
Which version did you last test?

**Comment by Shea690901** [comment]
1.9.1....
I hope to be able to upgrade coming week (if there is a newer version in my distros repository)

**Comment by albert-github** [comment]
Well in the 1.9.1 version there was a small problem regarding the files  send into the clang parser, this has been fixed by means of the pull request #9073 (based ion the issue #8784). This is available in the current master, though it is unlikely that this will be available in any distr0.

**Comment by Shea690901** [comment]
Ok, I'll try to take a look, at worst I might write my own package...

**Comment by Shea690901** [comment]
?!?
~~~
# ./bin/doxygen --version 
1.9.4 (a85850bf5b32843d6cde6b9a25b147c9a40bbc8a)
~~~
the above error is corrected, **but**:
~~~
...
Parsing files
Preprocessing src/main.cc...
error: clang: Failed to parse translation unit src/main.cc
...
~~~
when at the same time `src/main.cc` compiles without a single warning!!!
Also the files within `src` themselves, while correctly documented (*even main.cc*), are missing from the documentation....
Something still working with version 1.9.3...

**Comment by albert-github** [comment]
Good that the original issue is solved.

Regarding the `main.cc` it is hard to tel what is wrong as we have no information about the contents of `main.cc` and the settings in the doxygen settings file so:
- As the original issue corrected, please create a new issue for this issue
- Can you please attach there a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- Please also specify the **full** doxygen version used (`doxygen -v`). Will be `1.9.4 (a85850bf5b32843d6cde6b9a25b147c9a40bbc8a)`

**Comment by Shea690901** [comment]
Apparently I forgot to update the config file....

Now I've left "merely" to write a package, so I might install this doxygen globaly on my system 😉

**Comment by albert-github** [comment]
Reopening issue, please don't close the issue as this will be done at the moment of an official release.

**Comment by Shea690901** [comment]
👍

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
